waku-bindings/src/node/relay.rs

```diff
@@ -191,55 +162,51 @@ pub fn waku_enough_peers(pubsub_topic: Option<WakuPubSubTopic>) -> Result<bool>
     handle_response(code, &result)
 }
 
-pub fn waku_relay_subscribe(pubsub_topic: Option<WakuPubSubTopic>) -> Result<()> {
-    let pubsub_topic = pubsub_topic
-        .unwrap_or_else(waku_default_pubsub_topic)
-        .to_string();
-
-    let pubsub_topic_ptr = CString::new(pubsub_topic)
-        .expect("CString should build properly from pubsub topic")
-        .into_raw();
-
+pub fn waku_relay_subscribe(content_filter: &ContentFilter) -> Result<()> {
+    let content_filter_ptr = CString::new(
+        serde_json::to_string(content_filter)
+            .expect("ContentFilter should always succeed to serialize"),
+    )
+    .expect("ContentFilter should always be able to be serialized")
+    .into_raw();
     let mut error: String = Default::default();
     let error_cb = |v: &str| error = v.to_string();
     let code = unsafe {
         let mut closure = error_cb;
         let cb = get_trampoline(&closure);
         let out = waku_sys::waku_relay_subscribe(
-            pubsub_topic_ptr,
+            content_filter_ptr,
             cb,
             &mut closure as *mut _ as *mut c_void,
         );
 
-        drop(CString::from_raw(pubsub_topic_ptr));
+        drop(CString::from_raw(content_filter_ptr));
 
         out
     };
 
     handle_no_response(code, &error)
 }
```

Comment:
I feel like there must be a way to make this generic to avoid the repeated code. :thinking: